system_setup/server/controllers/locale.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from system_setup.common.helpers import get_windows_locale
+from subiquity.server.controllers.locale import LocaleController
+
+
+class LocaleController(LocaleController):
+
+    def start(self):
+        win_lang = get_windows_locale()
+        self.model.selected_language = os.environ.get("LANG") \
+            or self.autoinstall_default
+        if win_lang is not None:
+            self.model.selected_language = win_lang + ".UTF-8"
```

Comment:
We should verify that the locale is actually supported and enabled. 
This can be done by checking if the locale exists in /etc/locale.gen and is not commented out.

If it exists and is commented out, then remove the comment and run locale-gen at the end of the configuration process. This should done in the other branch on the "commit changes at the end"

Reply:
Created a card in Jira